<subsystem_configuration>
    <import package="rtt_barrett_hand_msgs" />
    <import package="barrett_hand_move_action" />
    <import package="eigen_typekit" />
    <import package="rtt_control_msgs" />
    <import package="rtt_rosclock" />
    <import package="controller_common" />
    <import package="velma_controller" />
    <import package="rtt_cartesian_trajectory_msgs" />
    <import package="rtt_std_msgs" />
    <import package="rtt_tf" />
    <import package="port_operations" />
    <import package="force_control" />
    <import package="rtt_force_control_msgs" />
    <import package="rtt_velma_core_cs_task_cs_msgs" />
    <import package="rtt_barrett_hand_action_msgs" />
    <import package="rtt_grasped_action_msgs" />
    <import package="rtt_identification_action_msgs" />
    <import package="grasped_action" />
    <import package="identification_action" />
    <import package="visual_servo_end_effector" />

    <io_buffer alias="st" name="velma_core_st3" />
    <io_buffer alias="cmd" name="velma_core_cmd3" />

    <!--<component type="visual_servo_end_effector_types::VisualServoComponent" name="visual_servo" />-->
    <component type="visual_servo_end_effector_types::ExtCommandsComponent" name="ext_commands" running="true" />
    <component type="visual_servo_end_effector_types::ManipulationStrategyComponent" name="manipulation_strategy" />
    <component type="visual_servo_end_effector_types::VisualServoFakeComponent" name="visual_servo_fake" />

    <!--<connection from="stSplit.rWristPose_OUTPORT"       to="visual_servo.st_rWristPose_INPORT" />
    <connection from="stSplit.lWristPose_OUTPORT"       to="visual_servo.st_lWristPose_INPORT" />
    <connection from="visual_servo.cmd_cart_r_pose_OUTPORT"    to="cmdConcate.cart_r_pose_INPORT" />
    <connection from="visual_servo.cmd_cart_l_pose_OUTPORT"    to="cmdConcate.cart_l_pose_INPORT" />

    <connection from="stSplit.q_OUTPORT"             to="visual_servo.q_all_INPORT" />
    <connection from="stSplit.rArm_t_OUTPORT"        to="visual_servo.t_rArm_INPORT" />
    <connection from="stSplit.lArm_t_OUTPORT"        to="visual_servo.t_lArm_INPORT" />
    <connection from="stSplit.torso_t_OUTPORT"        to="visual_servo.t_torso_INPORT" />-->


    <connection from="stSplit.rWristPose_OUTPORT"       to="visual_servo_fake.st_rWristPose_INPORT" />
    <connection from="stSplit.lWristPose_OUTPORT"       to="visual_servo_fake.st_lWristPose_INPORT" />
    <connection from="visual_servo_fake.cmd_cart_r_pose_OUTPORT"    to="cmdConcate.cart_r_pose_INPORT" />
    <connection from="visual_servo_fake.cmd_cart_l_pose_OUTPORT"    to="cmdConcate.cart_l_pose_INPORT" />

    <connection from="stSplit.q_OUTPORT"             to="visual_servo_fake.q_all_INPORT" />
    <connection from="stSplit.rArm_t_OUTPORT"        to="visual_servo_fake.t_rArm_INPORT" />
    <connection from="stSplit.lArm_t_OUTPORT"        to="visual_servo_fake.t_lArm_INPORT" />
    <connection from="stSplit.torso_t_OUTPORT"        to="visual_servo_fake.t_torso_INPORT" />

    <connection from="stSplit.rWristPose_OUTPORT"       to="manipulation_strategy.st_rWristPose_INPORT" />
    <connection from="stSplit.lWristPose_OUTPORT"       to="manipulation_strategy.st_lWristPose_INPORT" />
    <connection from="visual_servo_fake.xce_end_OUTPORT"    to="manipulation_strategy.xce_end_INPORT" />


    <connection from="manipulation_strategy.cmd_cart_r_imp_OUTPORT"       to="cmdConcate.cart_r_imp_INPORT" />


    <ros_stream port="ext_commands.cmd_INPORT"    topic="/visual_servo_end_effector/cmd_in" />

    <!-- debug streams -->
    <ros_stream port="stSplit._OUTPORT"    topic="/visual_servo_end_effector/st" />
    <ros_stream port="cmdConcate.msg_OUTPORT"   topic="/visual_servo_end_effector/cmd_out" />
    <!--<ros_stream port="visual_servo.pose_array_OUTPORT"   topic="/visual_servo_end_effector/pose_array" />-->

    <ros_stream port="visual_servo_fake.cmd_cart_r_pose_INPORT"    topic="/visual_servo_end_effector/trajectory_in" />

    <ros_stream port="manipulation_strategy.strategy_mode_OUTPORT"    topic="/visual_servo_end_effector/strategy_mode_out" />

</subsystem_configuration>
